collie/model/hybrid_pretrained_matrix_factorization.py

```diff
@@ -309,34 +374,32 @@ def forward(self, users: torch.tensor, items: torch.tensor) -> torch.tensor:
 
         """
 
-        if ((self.item_metadata is not None and str(self.device) != str(self.item_metadata.device))
-            or (self.user_metadata is not None
-                and str(self.device) != str(self.user_metadata.device))):
+        if (
+            self.item_metadata is not None
+            and str(self.device) != str(self.item_metadata.device)
+        ) or (
+            self.user_metadata is not None
+            and str(self.device) != str(self.user_metadata.device)
+        ):
             self._move_any_external_data_to_device()
 
         if self.item_metadata is not None and self.user_metadata is not None:
-            item_metadata_output = self.item_metadata[items, :]
-            if self.item_metadata_layers is not None:
-                for metadata_nn_layer in self.item_metadata_layers:
-                    item_metadata_output = self.dropout(
-                        F.leaky_relu(
-                            metadata_nn_layer(item_metadata_output)
-                        )
-                    )
+            self._calculate_metadata_output(
+                metadata_type='user',
+                users=users,
+                items=items
+            )
 
-            user_metadata_output = self.user_metadata[users, :]
-            if self.user_metadata_layers is not None:
-                for metadata_nn_layer in self.user_metadata_layers:
-                    user_metadata_output = self.dropout(
-                        F.leaky_relu(
-                            metadata_nn_layer(user_metadata_output)
-                        )
-                    )
+            self._calculate_metadata_output(
+                metadata_type='item',
+                users=users,
+                items=items
+            )
 
             combined_output = torch.cat((self.embeddings[0](users),
                                          self.embeddings[1](items),
-                                         item_metadata_output,
-                                         user_metadata_output), 1)
+                                         self.item_metadata_output,
+                                         self.user_metadata_output), 1)
```

Comment:
Ditto here and below about the ordering.

Reply:
👍